src/error.rs

```diff
@@ -23,28 +23,26 @@ pub enum Error {
 }
 
 impl Error {
+    #[deprecated(since = "0.0.22", note = "Please use the to_string() method instead")]
     pub fn description(&self) -> &str {
         match *self {
             #[cfg(feature = "std")]
-            Error::IO(_) => { "IO error" }
-            Error::Scroll(_) => { "Scroll error" }
-            Error::BadMagic(_) => { "Invalid magic number" }
-            Error::Malformed(_) => { "Entity is malformed in some way" }
+            Error::IO(_) => "IO error",
+            Error::Scroll(_) => "Scroll error",
+            Error::BadMagic(_) => "Invalid magic number",
+            Error::Malformed(_) => "Entity is malformed in some way",
         }
     }
 }
 
 #[cfg(feature = "std")]
 impl error::Error for Error {
-    fn description(&self) -> &str {
-        Error::description(self)
-    }
-    fn cause(&self) -> Option<&error::Error> {
+    fn source(&self) -> Option<&(dyn error::Error + 'static)> {
         match *self {
-            Error::IO(ref io) => { io.cause() }
-            Error::Scroll(ref scroll) => { scroll.cause() }
-            Error::BadMagic(_) => { None }
-            Error::Malformed(_) => { None }
+            Error::IO(ref io) => io.source(),
+            Error::Scroll(ref scroll) => scroll.source(),
```

Comment:
Neither `io::Error` nor `scroll::Error` implement `source`, and I don't expect they ever will, so this code does nothing. Let's just delete this method.

Reply:
Did you mean to use blanket implementation of `error::Error`?

Reply:
No. I don't think there is a blanket implementation? I meant to still implement `error::Error`, but use the provided default implementation for all methods. Note that I don't think there's anywhere in the crate that relies on the `error:Error` trait being implemented, but we still should. I don't know what best practices are since the description/cause changes, but as far as I can tell we're not gaining anything by implementing these methods currently.

Reply:
Or, you could change this `source` method to return references to the io and scroll errors, instead of delegating. Still not sure that is useful though. I'm open to other opinions.